ps_viewedproduct.php

```diff
@@ -236,6 +236,7 @@ protected function addViewedProduct($idProduct)
 
         if (!in_array($idProduct, $arr)) {
             $arr[] = $idProduct;
+            $arr = array_reverse(array_slice(array_reverse($arr), 0, (int) (Configuration::get('PRODUCTS_VIEWED_NBR') + 1)));
```

Comment:
![image](https://user-images.githubusercontent.com/14963751/158550110-2de2bf78-b8cb-4f96-9cda-c3aa0eab88d9.png)

@daresh see there

Reply:
@Progi1984 might help you on this

Reply:
Probably another pair of brackets would solve this issue, making it like this:
 $arr = array_reverse(array_slice(array_reverse($arr), 0, ((int) (Configuration::get('PRODUCTS_VIEWED_NBR')) + 1)));

Reply:
Fixed